src/ast/node.ts

```diff
@@ -18,15 +19,15 @@ export default class Node implements AstType {
   children: Node[];
   errors: ValidationError[] = [];
   lines: number[] = [];
-  type: string;
+  type: NodeType;
```

Comment:
It won't ever equal `node`, unless for whatever reason folks manually set the node type to `undefined`. Since `Node` is just a AST Node, users can technically set this value to whatever they want. But the Markdoc library will never set it to anything besides the existing `NodeType`, `error`, or `tag`.

Reply:
Then should we remove `'node'` from `NodeType`?

Reply:
Currently if you pass `undefined` to Node, then the type gets set to `node`. If we change that, that would be a breaking change, even if technically it probably never happens except for in bug cases. 

I think `Node.type === node` could just be our signature for "default Node"

Reply:
In what scenarios would one pass `undefined` when making a Node?

Reply:
We do this sometimes when we just want a blank node, which we fill in with attributes and children (and potentially type) later. Here is an example in the core codebase: https://github.com/markdoc/markdoc/blob/98dd88386e3982a10af8c76fc1c4bf8e30498318/src/ast/node.ts#L53

Reply:
makes sense

Reply:
I think it is confusing that `NodeType` is `node | tag | error | <all the Markdown node types>` but it is pretty small nit, and I cannot think of a better solution there since getting rid of `'node'` would bebreaking change

Reply:
I mean, the type that `NodeType` currently represents is literally the above, so feels to me like it makes sense to update the definition as such, instead of making it a subset. 

But I'm also fine making two definitions: one that is `NodeType` and one that is `NodeType | 'node' | 'tag' | 'error'`

Reply:
I would be + 1 on creating `MarkdownNodeType` and `NodeType` and a comment explaining `Node.type === "node" means a default Node